serializations.md

```diff
@@ -480,3 +481,23 @@ The channel offchain transaction is not included directly in the transaction tre
 , <closes_at>       , int()
 ]
 ```
+
+#### Proof of inclusion on state trees (POI)
+```
+[ {<accounts>  , [{binary(), [{binary(), [binary()]}]}]}
+, {<calls>     , [{binary(), [{binary(), [binary()]}]}]}
+, {<channels>  , [{binary(), [{binary(), [binary()]}]}]}
+, {<contracts> , [{binary(), [{binary(), [binary()]}]}]}
+, {<ns>        , [{binary(), [{binary(), [binary()]}]}]}
+, {<oracles>   , [{binary(), [{binary(), [binary()]}]}]}
+]
+```
+
+Each field is the representation of a subtree to the top level tree. The serialization should be interpreted as
+```
+{<tree>, [<proof_of_inclusion>]}
+where
+<proof_of_inclusion> := [{<root_hash>, [{<mpt_hash>, <mpt_value>}]}]
+```
+where `[{<mpt_hash>, <mpt_value>}]` is the sorted list of Merkel Patricia Tree nodes in the proof. If the subtree is empty,
```

Comment:
Can you please elaborate on the sorting? It appears to me [serialization is not sorted](https://github.com/aeternity/epoch/pull/1202/files#diff-033dadbc082ac1d0414b95ede7462cefR132) and [the verification function does not rely on sorting of `ProofDB`](https://github.com/aeternity/epoch/blob/6a320f1f9b75961145d6d7c77e761e0d3d096b3d/apps/aeutils/src/aeu_mp_trees.erl#L457-L460). Also, if proof of inclusion is "only" in a transaction (i.e. not in a state tree), for as long as all the signers sign the same binary, sorting is not strictly needed.

Reply:
The serialization is [sorted](http://erlang.org/doc/man/gb_trees.html#to_list-1) and the deserialization used to verify this, but this has been lost apparently. It is nice to have a canonical serialization when possible, and the sorting makes the serialization canonical.

Reply:
Thanks for the link, I had missed that.